TST: special: array types: test tol failure with `torch` backend

cc @mdhaber 
```
_________________________________________________________________________ test_support_alternative_backends[f_name_n_args9-torch] __________________________________________________________________________
scipy/special/tests/test_support_alternative_backends.py:24: in test_support_alternative_backends
    @given(data=strategies.data())
        f          = <function given.<locals>.run_test_as_given.<locals>.wrapped_test at 0x123cbe9e0>
        f_name_n_args = ('gammaln', 1)
        xp         = <module 'torch' from '/Users/lucascolley/mambaforge/envs/scipy-dev/lib/python3.10/site-packages/torch/__init__.py'>
scipy/special/tests/test_support_alternative_backends.py:51: in test_support_alternative_backends
    xp_assert_close(res, xp.asarray(ref, dtype=dtype_xp), rtol=eps**0.5, atol=eps*10,
        args_np    = [array(-10.)]
        args_xp    = [tensor(-10.0000, dtype=torch.float64)]
        data       = data(...)
        dtype      = 'float64'
        dtype_np   = <class 'numpy.float64'>
        dtype_xp   = torch.float64
        elements   = {'allow_subnormal': False, 'max_value': 10.0, 'min_value': -10.0}
        eps        = 2.220446049250313e-16
        f          = <function support_alternative_backends.<locals>.wrapped at 0x11f548c10>
        f_name     = 'gammaln'
        f_name_n_args = ('gammaln', 1)
        final_shape = ()
        mbs        = mutually_broadcastable_shapes(num_shapes=1)
        n_args     = 1
        ref        = array(5.80645377)
        res        = tensor(5.8065, dtype=torch.float64)
        shapes     = ((),)
        xp         = <module 'torch' from '/Users/lucascolley/mambaforge/envs/scipy-dev/lib/python3.10/site-packages/torch/__init__.py'>
scipy/_lib/_array_api.py:293: in xp_assert_close
    return xp.testing.assert_close(actual, desired, rtol=rtol, atol=atol,
E   AssertionError: Scalars are not close!
E   
E   Absolute difference: 1.0175748776930504e-07 (up to 2.220446049250313e-15 allowed)
E   Relative difference: 1.7524894158750067e-08 (up to 1.4901161193847656e-08 allowed)
E   Falsifying example: test_support_alternative_backends(
E       xp=<module 'torch' from '/Users/lucascolley/mambaforge/envs/scipy-dev/lib/python3.10/site-packages/torch/__init__.py'>,
E       f_name_n_args=('gammaln', 1),
E       data=data(...),
E   )
E   Draw 1: BroadcastableShapes(input_shapes=((),), result_shape=())
E   Draw 2: 'float64'
E   Draw 3: array(-10.)
E   Explanation:
E       These lines were always and only run by failing examples:
E           /Users/lucascolley/mambaforge/envs/scipy-dev/lib/python3.10/site-packages/torch/testing/_comparison.py:1000
E           /Users/lucascolley/mambaforge/envs/scipy-dev/lib/python3.10/site-packages/torch/testing/_comparison.py:1511
E           /Users/lucascolley/mambaforge/envs/scipy-dev/lib/python3.10/site-packages/torch/testing/_comparison.py:174
E           /Users/lucascolley/mambaforge/envs/scipy-dev/lib/python3.10/typing.py:308
E   
E   You can reproduce this example by temporarily adding @reproduce_failure('6.92.2', b'AAAAAAEAAAABgDMJDj////8=') as a decorator on your test case
        actual     = tensor(5.8065, dtype=torch.float64)
        atol       = 2.220446049250313e-15
        check_dtype = True
        check_namespace = True
        check_shape = True
        desired    = tensor(5.8065, dtype=torch.float64)
        err_msg    = None
        rtol       = 1.4901161193847656e-08
        xp         = <module 'scipy._lib.array_api_compat.torch' from '/Users/lucascolley/programming/scipy/build-install/lib/python3.10/site-packages/scipy/_lib/array_api_compat/torch/__init__.py'>
```

--- a/.github/workflows/array_api.yml
+++ b/.github/workflows/array_api.yml
@@ -91,7 +91,7 @@ jobs:
         export OMP_NUM_THREADS=2
         export SCIPY_USE_PROPACK=1
         # expand as new modules are added
-        # --array-api-backend == -b
-        python dev.py --no-build test -b pytorch -b numpy -b array_api_strict -s cluster -- --durations 3 --timeout=60
-        python dev.py --no-build test -b pytorch -b numpy -b array_api_strict -s fft -- --durations 3 --timeout=60
-        python dev.py --no-build test --array-api-backend all --tests scipy/_lib/tests/test_array_api.py
+        python dev.py --no-build test -b all -s cluster -- --durations 3 --timeout=60
+        python dev.py --no-build test -b all -s fft -- --durations 3 --timeout=60
+        python dev.py --no-build test -b all -t scipy.special.tests.test_support_alternative_backends -- --durations 3 --timeout=60
+        python dev.py --no-build test -b all -t scipy._lib.tests.test_array_api

--- a/scipy/special/tests/test_support_alternative_backends.py
+++ b/scipy/special/tests/test_support_alternative_backends.py
@@ -56,8 +56,8 @@ def test_support_alternative_backends(xp, data, f_name_n_args):
 
     eps = np.finfo(dtype).eps
     # PyTorch seems to struggle with precision near the poles of `gammaln`,
-    # so the tolerance needs to be quite loose (eps**0.2) - see gh-19935.
-    # To compensate, we aso check that that the root-mean-square error is
+    # so the tolerance needs to be quite loose (eps**0.2) - see gh-19935.
+    # To compensate, we also check that the root-mean-square error is
     # less than eps**0.5.
     ref = xp.asarray(ref, dtype=dtype_xp)
     xp_assert_close(res, ref, rtol=eps**0.2, atol=eps*10,

--- a/scipy/special/tests/test_support_alternative_backends.py
+++ b/scipy/special/tests/test_support_alternative_backends.py
@@ -55,5 +55,13 @@ def test_support_alternative_backends(xp, data, f_name_n_args):
     res = f(*args_xp)
 
     eps = np.finfo(dtype).eps
-    xp_assert_close(res, xp.asarray(ref, dtype=dtype_xp), rtol=eps**0.5, atol=eps*10,
-                    check_namespace=True, check_shape=True, check_dtype=True)
+    # PyTorch seems to struggle with precision near the poles of `gammaln`,
+    # so the tolerance needs to be quite loose (eps**0.2) - see gh-19935.
+    # To compensate, we aso check that that the root-mean-square error is
+    # less than eps**0.5.
+    ref = xp.asarray(ref, dtype=dtype_xp)
+    xp_assert_close(res, ref, rtol=eps**0.2, atol=eps*10,
+                    check_namespace=True, check_shape=True, check_dtype=True,)
+    xp_assert_close(xp.sqrt(xp.mean(res**2)), xp.sqrt(xp.mean(ref**2)),
+                    rtol=eps**0.5, atol=eps*10,
+                    check_namespace=False, check_shape=False, check_dtype=False,)

